mautic/mautic
# Fix Doctrine metadata cache handling during installation

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | 4.x
| Bug fix?                               | yes
| New feature?                           | no
| Deprecations?                          | no
| BC breaks?                             | no
| Automated tests included?              | no
| Related user documentation PR URL      | N/A
| Related developer documentation PR URL | N/A
| Issue(s) addressed                     | Fixes #10480 




#### Description:

As described in #10480, we currently remove the cache folder during the installation process. More specifically, just before the database is installed. This approach is problematic because it breaks things like Symfony's Dependency Injection, causing invalid database credentials and/or connection errors to completely crash the installer. Luckily, [Doctrine has an option to only remove the metadata cache](https://www.doctrine-project.org/projects/doctrine-orm/en/2.9/reference/caching.html#clearing-the-cache), which is all we need (more context in https://github.com/mautic/mautic/pull/10481#issuecomment-930137452). This PR changes our logic so that we only clear the metadata cache. This fixes #10480 and significantly improves Mautic's installation speed.



#### Steps to test this PR:
1. Load up this PR locally
2. For the remainder of the steps, we assume that you're using DDEV for local development. This allows you to easily drop the database and start clean
3. Before testing each scenario below, please ensure you have a clean state where Mautic is not installed: `ddev delete --omit-snapshot --yes && rm -rf var/cache && rm app/config/local.php && ddev start`

**Scenarios to test**:

For testing the **CLI installation**, please run `ddev exec php bin/console mautic:install https://mautic.ddev.site --db_driver=pdo_mysql --db_host=db --db_user=db --db_name=db --db_password=db --db_port=3306 --db_table_prefix="mau_" --admin_password=mautic --admin_email="demo@ddev.local"` - simply remove `--db_table_prefix="mau_"` if you want to test without a prefix.

|Type| Database info | Expected outcome
|--|--|--|
|UI installation| Wrong credentials, NO prefix | ❌
|UI installation| Wrong credentials, WITH prefix | ❌
|UI installation| Correct credentials, NO prefix | ✅
|UI installation| Correct credentials, WITH prefix | ✅
|CLI installation| Wrong credentials, NO prefix | ❌
|CLI installation| Wrong credentials, WITH prefix | ❌
|CLI installation| Correct credentials, NO prefix | ✅
|CLI installation| Correct credentials, WITH prefix | ✅
